File: src/main/java/org/tensorflow/keras/initializers/Ones.java
package org.tensorflow.keras.initializers;

import org.tensorflow.Operand;
import org.tensorflow.op.Ops;
import org.tensorflow.types.TInt32;
import org.tensorflow.types.family.TNumber;

public class Ones extends Initializer {
    @Override
    public <T extends TNumber> Operand<T> initialize(Ops tf, Operand<TInt32> shape, Class<T> dtype) {
//        return tf.fill(shape, tf.constant(1.0f, dtype));
        return tf.fill(shape, tf.dtypes.cast(tf.constant(1.0f), dtype));
    }
}

File: src/main/scala/org/tensorflow/keras/initializers/Ones.scala
package org.tensorflow.keras.initializers

import org.tensorflow.Operand
import org.tensorflow.op.Ops
import org.tensorflow.types.TInt32
import org.tensorflow.types.family.TNumber

class Ones extends Initializer {
  override def initialize[T <: TNumber](tf: Ops, shape: Operand[TInt32], dtype: Class[T]): Operand[T] =
    tf.fill(shape, tf.dtypes.cast(tf.constant(1.0f), dtype))
}
